Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m

```diff
@@ -3766,11 +3685,10 @@ - (void)dealloc
 	[self removeObserver:self forKeyPath:SPFavoriteSSLCertificateFileLocationEnabledKey];
 	[self removeObserver:self forKeyPath:SPFavoriteSSLCertificateFileLocationKey];
 	[self removeObserver:self forKeyPath:SPFavoriteSSLCACertFileLocationEnabledKey];
-	[self removeObserver:self forKeyPath:SPFavoriteSSLCACertFileLocationKey];
+    [self removeObserver:self forKeyPath:SPFavoriteSSLCACertFileLocationKey];
+    [self removeObserver:self forKeyPath:SPBookmarksChangedNotification];
 
-	for(NSURL *url in resolvedBookmarks){
-		[url stopAccessingSecurityScopedResource];
-	}
+    [SecureBookmarkManager.sharedInstance stopAllSecurityScopedAccess];
```

Comment:
Is there a chance that we call this from somewhere and stop access while somewhere else is using a bookmark? Likely not, but perhaps we should count calls to request and calls to stop?

Reply:
will review the code and see where we request and revoke access.

Reply:
> Likely not, but perhaps we should count calls to request and calls to stop?
We sort of already do ... the `SecureBookmarkManager.sharedInstance.resolvedBookmarks` array contains current bookmarks we have requested access to...